x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation_logic.ts

```diff
@@ -0,0 +1,127 @@
+  setRawName(rawName: string): { rawName: string };
+  setSelectedIndexedEngineNames(
+    selectedIndexedEngineNames: MetaEngineCreationValues['selectedIndexedEngineNames']
+  ): { selectedIndexedEngineNames: MetaEngineCreationValues['selectedIndexedEngineNames'] };
+  submitEngine(): void;
+}
+
+export const MetaEngineCreationLogic = kea<
+  MakeLogicType<MetaEngineCreationValues, MetaEngineCreationActions>
+>({
+  path: ['enterprise_search', 'app_search', 'meta_engine_creation_logic'],
+  actions: {
+    fetchIndexedEngineNames: (page = DEFAULT_META.page.current) => ({ page }),
+    onEngineCreationSuccess: true,
+    setIndexedEngineNames: (indexedEngineNames) => ({ indexedEngineNames }),
+    setRawName: (rawName) => ({ rawName }),
+    setSelectedIndexedEngineNames: (selectedIndexedEngineNames) => ({ selectedIndexedEngineNames }),
+    submitEngine: () => null,
+  },
+  reducers: {
+    indexedEngineNames: [
+      [],
+      {
+        setIndexedEngineNames: (_, { indexedEngineNames }) => indexedEngineNames,
+      },
+    ],
+    rawName: [
+      '',
+      {
+        setRawName: (_, { rawName }) => rawName,
+      },
+    ],
+    selectedIndexedEngineNames: [
+      [],
+      {
+        setSelectedIndexedEngineNames: (_, { selectedIndexedEngineNames }) =>
+          selectedIndexedEngineNames,
+      },
+    ],
+  },
+  selectors: ({ selectors }) => ({
+    name: [() => [selectors.rawName], (rawName: string) => formatApiName(rawName)],
+  }),
+  listeners: ({ values, actions }) => ({
+    fetchIndexedEngineNames: async ({ page }) => {
+      const { http } = HttpLogic.values;
+      let response: { results: EngineDetails[]; meta: Meta } | undefined;
+
+      try {
+        response = await http.get('/api/app_search/engines', {
+          query: { type: 'indexed', 'page[current]': page, 'page[size]': DEFAULT_META.page.size },
+        });
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+
+      if (response) {
+        const engineNames = response.results.map((result) => result.name);
+        actions.setIndexedEngineNames([...values.indexedEngineNames, ...engineNames]);
+
+        if (page < response.meta.page.total_pages) {
+          actions.fetchIndexedEngineNames(page + 1);
+        }
+      }
+    },
+    onEngineCreationSuccess: () => {
+      const { name } = values;
+      const { navigateToUrl } = KibanaLogic.values;
+      const enginePath = generatePath(ENGINE_PATH, { engineName: name });
+
+      setQueuedSuccessMessage(META_ENGINE_CREATION_SUCCESS_MESSAGE);
+      navigateToUrl(enginePath);
+    },
+    submitEngine: async () => {
+      const { http } = HttpLogic.values;
+      const { name, selectedIndexedEngineNames } = values;
+
+      const body = JSON.stringify({
+        name,
+        type: 'meta',
+        source_engines: selectedIndexedEngineNames,
+      });
+
+      try {
+        const response = await http.post('/api/app_search/engines', { body });
```

Comment:
Shoot sorry, I hadn't realized in my original suggestion that we weren't using data from the API response

```suggestion
        await http.post('/api/app_search/engines', { body });
```